packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1853,6 +1871,29 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         this.dataStores.processAliasMessage(message, localOpMetadata, local);
     }
 
+    /**
+     * Emits the Signal event and update the perf signal data.
+     * @param clientSignalSequenceNumber - is the client signal sequence number to be uploaded.
+     */
+    private emitSignalEvent(clientSignalSequenceNumber: number) {
+        const currentSequence = Math.floor(clientSignalSequenceNumber / this.defaultTelemetrySignalSampleCount);
+        // Check to see if we missed the previous sequence number.
+        if (this._perfSignalData.expectedSignalSequenceFloorNumber !== currentSequence) {
```

Comment:
I think you are checking that every 100 signals we will get here. And that capture if our signal was lost.
The suggestion below (in another comment) would capture that (lost signals), but also reordered signals, which is a bit stronger (assuming we should not see reordering by design of the system)